addons/sale_mrp/models/sale_mrp.py

```diff
@@ -35,14 +35,15 @@ def _compute_qty_delivered(self):
                         (b.product_tmpl_id == order_line.product_id.product_tmpl_id and not b.product_id)))
                 if relevant_bom:
                     # In case of dropship, we use a 'all or nothing' policy since 'bom_line_id' was
-                    # not written on a move coming from a PO.
+                    # not written on a move coming from a PO: all moves must be done and the last
+                    # one must be a delivery to the customer (to avoid the return case)
                     # FIXME: if the components of a kit have different suppliers, multiple PO
                     # are generated. If one PO is confirmed and all the others are in draft, receiving
                     # the products for this PO will set the qty_delivered. We might need to check the
                     # state of all PO as well... but sale_mrp doesn't depend on purchase.
                     if dropship:
-                        moves = order_line.move_ids.filtered(lambda m: m.state != 'cancel')
-                        if moves and all([m.state == 'done' for m in moves]):
+                        moves = order_line.move_ids.filtered(lambda m: m.state == 'done').sorted(key='id')
+                        if moves and moves[-1:].location_dest_id.usage == 'customer':
```

Comment:
This unfortunately won't work if you have backorders or multiple pickings (i.e. you are ignoring non-done+non-cancelled moves, which breaks the "all or nothing" policy) :cry:

Also a test would be nice :) especially one that can include this case too: https://github.com/odoo/odoo/pull/73372 Of course it will not match the use case exactly since purchase isn't part of this module, but you can probably hack similar enough behavior.